### PR TITLE
Add staked capacity test

### PR DESCRIPTION
**Pull request by Kishan-Dhakan**

### CODE REVIEW

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | test/staked-capacity  |
| gosdk  | staging  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | fix/stakepool_test  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0dns  | staging  |

File: tests/cli_tests/blobber_staked_capacity_test.go [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+)
+
+const (
+	newStakeWallet = "newStakeWallet"
+)
+
+func TestStakePool(testSetup *testing.T) {
+	t := test.NewSystemTest(testSetup)
+
+	_, err := createWallet(t, configPath)
```
Comment (stewartie4) [inline tests/cli_tests/blobber_staked_capacity_test.go:22]: this test setup should now be wrapped in a t.TestSetup block like the other tests

File: tests/cli_tests/blobber_staked_capacity_test.go [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+	require.Nil(t, err, "Error registering wallet", err)
+
+	// get the list of blobbers
+	blobbersList := getBlobbersList(t)
+	require.Greater(t, len(blobbersList), 0, "No blobbers found")
+
+	t.RunSequentiallyWithTimeout("Total stake in a blobber can never be less than it's used capacity", 800*time.Minute, func(t *test.SystemTest) {
+		_, err := createWallet(t, configPath)
```
Comment (stewartie4) [inline tests/cli_tests/blobber_staked_capacity_test.go:30]: Looks like really good/complex work, there is a lot going on in this test and it is 200 lines long, can some logic be broken out into functions to make it more readable? 
eg. anywhere there is a comment could probably be a function call

**Review by stewartie4 — CHANGES_REQUESTED** [review]

File: tests/cli_tests/zboxcli_stake_unstake_token_test.go [hunk 3]
```diff
@@ -297,8 +300,11 @@ func stakePoolInfo(t *test.SystemTest, cliConfigFilename, params string) ([]stri
 }
 
 func unstakeTokens(t *test.SystemTest, cliConfigFilename, params string) ([]string, error) {
-	t.Log("Unlocking tokens from stake pool...")
-	return cliutils.RunCommand(t, fmt.Sprintf("./zbox sp-unlock %s --silent --wallet %s_wallet.json --configDir ./config --config %s", params, escapedTestName(t), cliConfigFilename), 3, time.Second*2)
+	return unstakeTokensForWallet(t, cliConfigFilename, escapedTestName(t), params)
+}
+
+func unstakeTokensForWallet(t *test.SystemTest, cliConfigFilename, wallet, params string) ([]string, error) {
+	return cliutils.RunCommand(t, fmt.Sprintf("./zbox sp-unlock %s --silent --wallet %s_wallet.json --configDir ./config --config %s", params, wallet, cliConfigFilename), 3, time.Second*2)
```
Comment (stewartie4) [inline tests/cli_tests/zboxcli_stake_unstake_token_test.go:307]: re-add the log line
t.Log("Unlocking tokens from stake pool...")
here

File: tests/cli_tests/zboxcli_stake_unstake_token_test.go [hunk 4]
```diff
@@ -282,8 +282,11 @@ func getTransaction(t *test.SystemTest, cliConfigFilename, params string) ([]str
 }
 
 func stakeTokens(t *test.SystemTest, cliConfigFilename, params string, retry bool) ([]string, error) {
-	t.Log("Staking tokens...")
-	cmd := fmt.Sprintf("./zbox sp-lock %s --silent --wallet %s_wallet.json --configDir ./config --config %s", params, escapedTestName(t), cliConfigFilename)
+	return stakeTokensForWallet(t, cliConfigFilename, escapedTestName(t), params, retry)
+}
+
+func stakeTokensForWallet(t *test.SystemTest, cliConfigFilename, wallet, params string, retry bool) ([]string, error) {
+	cmd := fmt.Sprintf("./zbox sp-lock %s --silent --wallet %s_wallet.json --configDir ./config --config %s", params, wallet, cliConfigFilename)
```
Comment (stewartie4) [inline tests/cli_tests/zboxcli_stake_unstake_token_test.go:289]: re-add the log line
t.Log("Staking tokens...")
here

**Review by stewartie4 — CHANGES_REQUESTED** [review]

File: tests/cli_tests/blobber_staked_capacity_test.go (hunk 2 above)
Comment (stewartie4) [reply tests/cli_tests/blobber_staked_capacity_test.go:34]: there is still a lot of logic that could be broken into functions here
eg.

fetchStakePoolInfo
stakeBlobbers
assertStakeGreaterThanUsedCapacity
createAllocationOfMaxSizeBlobbersCanHonour
...
....
.....
unstakeTokens

File: tests/cli_tests/blobber_staked_capacity_test.go [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+const (
+	newStakeWallet = "newStakeWallet"
+)
+
+func TestStakePool(testSetup *testing.T) {
+	t := test.NewSystemTest(testSetup)
+
+	var blobbersList []climodel.BlobberInfo
+
+	t.TestSetup("stake_pool_test", func() {
```
Comment (stewartie4) [inline tests/cli_tests/blobber_staked_capacity_test.go:24]: This label should be something like "register wallet and get blobbers"

File: tests/cli_tests/blobber_staked_capacity_test.go [hunk 6]
```diff
@@ -0,0 +1,272 @@
+package cli_tests
```
Comment (stewartie4) [inline tests/cli_tests/blobber_staked_capacity_test.go:1]: If these tests are sequential then they should probably run at the beginning of the suite, so a "0_" should be appended to the file name

**Review by stewartie4 — CHANGES_REQUESTED** [review]
few final comments

**Review by stewartie4 — APPROVED** [review]

**Review by stewartie4 — APPROVED** [review]
Thanks @Jayashsatolia403 looks good

**Pull request merged**